File: DesignPatternsExercises/java/StrategyDesignPattern/Solution/TSVExtraction.java
class TSVExtraction implements Extraction {
    @Override
    public void extract() {
        // __________(12)__________;
        open();
        System.out.println("TSV Data type is getting extracted by separating row by tabs...");
        // __________(13)__________;
        close();
    }

    @Override
    public void open() {
        System.out.println("Opening TSV file...");
    }

    @Override
    public void close() {
        System.out.println("Closing TSV file...");
    }
}

File: DesignPatternsExercises/ruby/strategy_design_pattern/solution/tsv_extraction.rb
require_relative 'extraction'

class TSVExtraction < Extraction
  def extract
    # __________(8)__________
    open
    puts 'TSV Data type is getting extracted by separating row by tabs...'
    # __________(9)__________
    close
  end

  def open
    puts 'Opening TSV file...'
  end

  def close
    puts 'Closing TSV file...'
  end
end
